osu.Game/Graphics/Containers/BeatSyncedContainer.cs

```diff
@@ -94,6 +94,9 @@ protected override void Update()
 
             TimeSinceLastBeat = beatLength - TimeUntilNextBeat;
 
+            if (currentTrackTime < TimeSinceLastBeat)
```

Comment:
This conditional doesn't match with what you say it's doing in the PR description. You're comparing track time to... time since last?

Also we require test coverage for changes like this. I'll look at fixing this up.

Reply:
In this context, I understand `TimeSinceLastBeat` as the delay between the current beat that we are about to fire `OnNewBeat` on, and the actual firing of `OnNewBeat`. So, if this delay is longer than the time elapsed since track start, this beat must precede track start.

Reply:
I'm still not sure on this. I've opened a PR with an alternative to your take, with testing and stuff #13894.